Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -109,43 +87,43 @@ public class CRTClientEngine: HttpClientEngine {
         }
     }
     
-    public func makeHttpRequestStreamOptions(_ request: SdkHttpRequest) -> (HttpRequestOptions, Future<HttpResponse>) {
-        let future = Future<HttpResponse>()
-        let crtRequest = request.toHttpRequest(bufferSize: windowSize)
+    public func makeHttpRequestStreamOptions(_ request: SdkHttpRequest, _ continuation: StreamContinuation) -> HttpRequestOptions {
         let response = HttpResponse()
-        
+        let crtRequest = request.toHttpRequest(bufferSize: windowSize)
         let streamReader: StreamReader = DataStreamReader()
         
         let requestOptions = HttpRequestOptions(request: crtRequest) { [self] (stream, _, httpHeaders) in
             logger.debug("headers were received")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.getResponseStatusCode()))
-                ?? HttpStatusCode.notFound
+            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode))
+            ?? HttpStatusCode.notFound
             response.headers.addAll(httpHeaders: httpHeaders)
         } onIncomingHeadersBlockDone: { [self] (stream, _) in
             logger.debug("header block is done")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.getResponseStatusCode()))
-                ?? HttpStatusCode.notFound
+            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode))
+            ?? HttpStatusCode.notFound
         } onIncomingBody: { [self] (_, data) in
             logger.debug("incoming data")
             
             let byteBuffer = ByteBuffer(data: data)
             streamReader.write(buffer: byteBuffer)
-        } onStreamComplete: { [self] (_, error) in
+        } onStreamComplete: { [self] (stream, error) in
             logger.debug("stream completed")
             streamReader.hasFinishedWriting = true
             if case let CRTError.crtError(unwrappedError) = error {
                 if unwrappedError.errorCode != 0 {
                     logger.error("Response encountered an error: \(error)")
                     streamReader.onError(error: ClientError.crtError(error))
-                    future.fail(error)
-                    return
+                    continuation.resume(throwing: error)
                 }
             }
-
+            
             response.body = .stream(.reader(streamReader))
-            future.fulfill(response)
+            self.logger.debug("response came back with success")
```

Comment:
Putting this log here seems like it would create a ton of noise.  I suspect you meant to delete this?